sw/device/lib/testing/otp_ctrl_testutils.c

```diff
@@ -0,0 +1,43 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/*
+ * OTP the Direct Access Interface (DAI) operation time-out in micro seconds.
+ *
+ * It is not possible to predict the specific cycle count that a DAI operation
+ * takes, thus arbitrary value of 100us is used.
+ */
+const uint8_t kOtpDaiTimeoutUs = 100;
+
+/**
+ * Checks whether the DAI operation has finished.
+ */
+static bool dai_finished(const otp_ctrl_t *otp) {
+  dif_otp_ctrl_status_t status;
+  CHECK_DIF_OK(dif_otp_ctrl_get_status(&otp->dif, &status));
+  return bitfield_bit32_read(status.codes, kDifOtpCtrlStatusCodeDaiIdle);
+}
+
+otp_ctrl_t otp_ctrl_init(void) {
+  otp_ctrl_t otp;
+
+  mmio_region_t region =
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR);
+  CHECK_DIF_OK(dif_otp_ctrl_init(region, &otp.dif));
+
+  return otp;
```

Comment:
I think we need to set and follow a guideline - for other IPs we made the decision to let the top level test prepare the DIF handle so that the testutils are also portable across chip implementations. I kinda like going with that approach. 

Lets not have testutils depend on a specific top level.

Reply:
I agree, (like the DIFs, it would be nice if the testutils were toplevel agnostic). However, there were a couple testutils that slipped through that are toplevel specific, so that was probably confusing! Sorry about that @silvestrst.

Anywho, I don't think think this should even be a testutil function (as a I stated above). So this comment won't even apply.

Reply:
Done.